### PR TITLE
static-site: Remove sharp-related dependencies

### LINKED ISSUE #597: static-site unbuildable under nodejs v16, v18 on darwin-arm64 aka M1
### Current Behavior

Current master (d35785ed) under nodejs v16 or v18 (I can't conda install v14 on M1 architecture, but that's another matter)

The current setup is working on CI and on heroku, so it's not blocking. But it's a development pain. I'll try to debug, but posting here in case others have solutions!

```
$ npm run build --build-static

npm ERR! code 1
npm ERR! path /Users/enigma/projects/nextstrain/nextstrain.org/static-site/node_modules/sharp
npm ERR! command failed
npm ERR! command sh /var/folders/7w/rvnm5fgd7kb83fw8lg882x680000gn/T/install-23ce1806.sh
npm ERR! CC(target) Release/obj.target/nothing/../node-addon-api/src/nothing.o
npm ERR!   LIBTOOL-STATIC Release/nothing.a
npm ERR!   TOUCH Release/obj.target/libvips-cpp.stamp
npm ERR!   CXX(target) Release/obj.target/sharp/src/common.o
npm ERR! info sharp Downloading https://github.com/lovell/sharp-libvips/releases/download/v8.9.1/libvips-8.9.1-darwin-arm64v8.tar.gz
npm ERR! ERR! sharp Prebuilt libvips 8.9.1 binaries are not yet available for darwin-arm64v8
npm ERR! info sharp Attempting to build from source via node-gyp but this may fail due to the above error
npm ERR! info sharp Please see https://sharp.pixelplumbing.com/install for required dependencies
npm ERR! gyp info it worked if it ends with ok
npm ERR! gyp info using node-gyp@9.0.0
npm ERR! gyp info using node@16.17.0 | darwin | arm64
npm ERR! gyp info find Python using Python version 3.10.6 found at "/usr/local/opt/python@3.10/bin/python3.10"
npm ERR! gyp info spawn /usr/local/opt/python@3.10/bin/python3.10
npm ERR! gyp info spawn args [
npm ERR! gyp info spawn args   '/Users/enigma/miniconda3/envs/nextstrain.org/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
npm ERR! gyp info spawn args   'binding.gyp',
npm ERR! gyp info spawn args   '-f',
npm ERR! gyp info spawn args   'make',
npm ERR! gyp info spawn args   '-I',
npm ERR! gyp info spawn args   '/Users/enigma/projects/nextstrain/nextstrain.org/static-site/node_modules/sharp/build/config.gypi',
npm ERR! gyp info spawn args   '-I',
npm ERR! gyp info spawn args   '/Users/enigma/miniconda3/envs/nextstrain.org/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
npm ERR! gyp info spawn args   '-I',
npm ERR! gyp info spawn args   '/Users/enigma/Library/Caches/node-gyp/16.17.0/include/node/common.gypi',
npm ERR! gyp info spawn args   '-Dlibrary=shared_library',
npm ERR! gyp info spawn args   '-Dvisibility=default',
npm ERR! gyp info spawn args   '-Dnode_root_dir=/Users/enigma/Library/Caches/node-gyp/16.17.0',
npm ERR! gyp info spawn args   '-Dnode_gyp_dir=/Users/enigma/miniconda3/envs/nextstrain.org/lib/node_modules/npm/node_modules/node-gyp',
npm ERR! gyp info spawn args   '-Dnode_lib_file=/Users/enigma/Library/Caches/node-gyp/16.17.0/<(target_arch)/node.lib',
npm ERR! gyp info spawn args   '-Dmodule_root_dir=/Users/enigma/projects/nextstrain/nextstrain.org/static-site/node_modules/sharp',
npm ERR! gyp info spawn args   '-Dnode_engine=v8',
npm ERR! gyp info spawn args   '--depth=.',
npm ERR! gyp info spawn args   '--no-parallel',
npm ERR! gyp info spawn args   '--generator-output',
npm ERR! gyp info spawn args   'build',
npm ERR! gyp info spawn args   '-Goutput_dir=.'
npm ERR! gyp info spawn args ]
npm ERR! gyp info spawn make
npm ERR! gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
npm ERR! warning: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: archive library: Release/nothing.a the table of contents is empty (no object file members in the library define global symbols)
npm ERR! ../src/common.cc:23:10: fatal error: 'vips/vips8' file not found
npm ERR! #include <vips/vips8>
npm ERR!          ^~~~~~~~~~~~
npm ERR! 1 error generated.
npm ERR! make: *** [Release/obj.target/sharp/src/common.o] Error 1
npm ERR! gyp ERR! build error 
npm ERR! gyp ERR! stack Error: `make` failed with exit code: 2
npm ERR! gyp ERR! stack     at ChildProcess.onExit (/Users/enigma/miniconda3/envs/nextstrain.org/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:194:23)
npm ERR! gyp ERR! stack     at ChildProcess.emit (node:events:513:28)
npm ERR! gyp ERR! stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:291:12)
npm ERR! gyp ERR! System Darwin 20.6.0
npm ERR! gyp ERR! command "/Users/enigma/miniconda3/envs/nextstrain.org/bin/node" "/Users/enigma/miniconda3/envs/nextstrain.org/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
npm ERR! gyp ERR! cwd /Users/enigma/projects/nextstrain/nextstrain.org/static-site/node_modules/sharp
npm ERR! gyp ERR! node -v v16.17.0
npm ERR! gyp ERR! node-gyp -v v9.0.0
npm ERR! gyp ERR! not ok

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/enigma/.npm/_logs/2022-09-28T02_24_23_381Z-debug-0.log

Build failed at ./build.sh line 37

You are responsible for clean up (sorry!)
```

### Expected behavior

Build succeeded 😂 

### How to reproduce

You may need a M1 mac to reproduce this?

### PR DESCRIPTION
**Pull request by victorlin**
## Description of proposed changes

These don't seem to be necessary, and make it harder to set up a dev environment on Apple silicon macs.

## Related issue(s)

- Resolves #597

## Checklist

- [ ] Checks pass
- [ ] Check that the removal of these gatsby plugins are acceptable
    - `gatsby-remark-images` ([thread](https://github.com/nextstrain/nextstrain.org/pull/754#discussion_r1448116570))

### CODE REVIEW

**Review by jameshadfield — COMMENTED** [review]
This is the only way I've found to build nextstrain.org on my current setup. It'd be great to get this merged!

> gatsby-remark-images seems to be actively used to scale down images such as [gisaid_vs_open_map.png](https://github.com/nextstrain/nextstrain.org/blob/4c91074c07ac7bd7843fda64b234aab70b714ff6/static-site/content/blog/img/gisaid_vs_open_map.png). This is something that can be done separately (and should be done in this PR).

I agree, and ideally via a script / command so that we can easily repeat it for any future images (assuming our next static-site approach doesn't do it automatically).

**Comment by victorlin** [comment]
@jameshadfield thanks for testing! I made this PR while working off of my personal computer where I don't want to install vips, and put it aside once I returned to using the work computer. I'll pick it up now that more devs are using it.

File: static-site/gatsby-config.js [hunk 1]
```diff
@@ -39,12 +39,6 @@ module.exports = {
       resolve: "gatsby-transformer-remark",
       options: {
         plugins: [
-          {
-            resolve: "gatsby-remark-images",
```
Comment (victorlin) [inline static-site/gatsby-config.js:43]: I found these alternatives to [gatsby-remark-images](https://www.gatsbyjs.com/plugins/gatsby-remark-images/), which I'll look into later:

- [gatsby-remark-images-anywhere](https://www.gatsbyjs.com/plugins/gatsby-remark-images-anywhere/)
- [gatsby-remark-relative-images](https://www.gatsbyjs.com/plugins/gatsby-remark-relative-images/)

Comment (victorlin) [reply static-site/gatsby-config.js:43]: Can't use either of those for this PR.

- [gatsby-remark-images-anywhere](https://www.gatsbyjs.com/plugins/gatsby-remark-images-anywhere/) requires gatsby-plugin-sharp.
- [gatsby-remark-relative-images](https://www.gatsbyjs.com/plugins/gatsby-remark-relative-images/) extends, not replaces, gatsby-remark-images.

Digging further, I don't think there is a way to serve Markdown images in Gatsby without `sharp`. All the different options in [this guide](https://www.gatsbyjs.com/docs/how-to/images-and-media/working-with-images-in-markdown/) end up using `gatsby-remark-images`/`gatsby-plugin-sharp`.

This is a dead end for the PR's intent to remove sharp-related dependencies.

**Comment by victorlin** [comment]
@jameshadfield due to https://github.com/nextstrain/nextstrain.org/pull/754#discussion_r1449528695, I'm going to close this PR. I'll keep the working branch in case you need it.

Alternatively, I've squashed the commits into 737539587f5a08eb2466ddd24153c30041e451e5 on branch [victorlin/patch/pr-754-remove-sharp-deps](https://github.com/nextstrain/nextstrain.org/tree/victorlin/patch/pr-754-remove-sharp-deps). If that works better, you can delete this PR's branch.

**Comment by jameshadfield** [comment]
> I'll keep the working branch in case you need it.

Thanks! I've deleted this as [I am now able to build sharp](https://github.com/nextstrain/nextstrain.org/issues/597#issuecomment-1888342859)

**Pull request closed without merging**